src/object/endpoint.c

```diff
@@ -226,7 +226,9 @@ void receiveIPC(tcb_t *thread, cap_t cap, bool_t isBlocking)
             if (do_call ||
                 seL4_Fault_get_seL4_FaultType(sender->tcbFault) != seL4_Fault_NullFault) {
                 if ((canGrant || canGrantReply) && replyPtr != NULL) {
-                    reply_push(sender, thread, replyPtr, sender->tcbSchedContext != NULL);
+                    bool_t canDonate = sender->tcbSchedContext != NULL
```

Comment:
this commit can also be merged, as long as it passes seL4test on all platforms. 

Ideally the benchmarks would run too as they stress test timeout faults more.